ddtrace/contrib/langchain/patch.py

```diff
@@ -0,0 +1,713 @@
+import os
+import sys
+from typing import TYPE_CHECKING
+
+import langchain
+from langchain.callbacks.openai_info import get_openai_token_cost_for_model
+
+from ddtrace import config
+from ddtrace.contrib._trace_utils_llm import BaseLLMIntegration
+from ddtrace.contrib.langchain.constants import text_embedding_models
+from ddtrace.contrib.langchain.constants import vectorstores
+from ddtrace.contrib.trace_utils import unwrap
+from ddtrace.contrib.trace_utils import with_traced_module
+from ddtrace.contrib.trace_utils import wrap
+from ddtrace.internal.agent import get_stats_url
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils.formats import asbool
+from ddtrace.pin import Pin
+
+
+if TYPE_CHECKING:
+    from ddtrace import Span
+
+
+log = get_logger(__name__)
+
+
+config._add(
+    "langchain",
+    {
+        "logs_enabled": asbool(os.getenv("DD_LANGCHAIN_LOGS_ENABLED", False)),
+        "metrics_enabled": asbool(os.getenv("DD_LANGCHAIN_METRICS_ENABLED", True)),
+        "span_prompt_completion_sample_rate": float(os.getenv("DD_LANGCHAIN_SPAN_PROMPT_COMPLETION_SAMPLE_RATE", 1.0)),
+        "log_prompt_completion_sample_rate": float(os.getenv("DD_LANGCHAIN_LOG_PROMPT_COMPLETION_SAMPLE_RATE", 0.1)),
+        "span_char_limit": int(os.getenv("DD_LANGCHAIN_SPAN_CHAR_LIMIT", 128)),
+        "_api_key": os.getenv("DD_API_KEY"),
+    },
+)
+
+
+class _LangChainIntegration(BaseLLMIntegration):
+    _integration_name = "langchain"
+
+    def __init__(self, config, stats_url, site, api_key):
+        super().__init__(config, stats_url, site, api_key)
+
+    def _set_base_span_tags(self, span):
+        # type: (Span) -> None
+        return None
+
+    def _logs_tags(self, span):
+        # type: (Span) -> str
+        api_key = span.get_tag("langchain.request.api_key") or ""
+        tags = "env:%s,version:%s,langchain.request.provider:%s,langchain.request.model:%s,langchain.request.type:%s,langchain.request.api_key:%s" % (  # noqa: E501
```

Comment:
Good question. For conformity purposes, having the tags always be present is helpful, especially for cases where (in the dashboard queries for example) we have to query for specific logs that do not have an empty tag value. I agree your recommendation would reduce the payload size, but I don't think this will effect payload sizes *that* much.